### ISSUE TITLE
Blackfriday adds extra blank line in code block within a list

### ISSUE THREAD

**Issue by cmbrandenburg** [russross/blackfriday#228]
Input:

```
* Foo

        bar

        qux
```

Expected output:

```
<ul>
<li><p>Foo</p>

<pre><code>bar

qux
</code></pre></li>
</ul>
```

Actual output:

```
<ul>
<li><p>Foo</p>

<pre><code>bar


qux
</code></pre></li>
</ul>
```

Notice the extra blank line between “bar” and “qux.”

**Comment by shurcooL** [comment]
Are you using `blackfriday.MarkdownCommon` or something else?

**Comment by cmbrandenburg** [comment]
I'm using Hugo, and I get the same result with `blackfriday-tool`. I can dig in to find out more details if you need them.

For what it's worth, the canonical Gruber Perl markdown program has the same bug.

**Comment by shurcooL** [comment]
I can reproduce.

> For what it's worth, the canonical Gruber Perl markdown program has the same bug.

That's helpful to know, thanks.

Interestingly, it also happens if using fenced code block:

```
* Foo
    
    ```     
    bar

    qux
    ```
```

**Comment by shurcooL** [comment]
Another test case:

```
* Foo
    
    ```     
    bar












    qux
    ```
```

That produces the same output as above.

**Comment by rtfb** [comment]
I think I have a fix. I'll test it a bit more and post later today.

**Comment by shurcooL** [comment]
Fixed via #229.
